### PR TITLE
dynamic date warning messages for incomes and adjustments

### PR DESCRIPTION
**Pull request by mdkaraman**
IVL-182771145

### CODE REVIEW

File: components/financial_assistance/app/views/financial_assistance/incomes/_other_income_form.html.erb [hunk 1]
```diff
@@ -1,54 +1,60 @@
...
+          <%= f.text_field :amount, placeholder: '0.00', value: number_to_currency(other_income.amount), required: true %>
         </div>
       </div>
-    <% end %>
-    <div class="col-md-3">
-      <%= f.label :frequency_kind, 'Amount *', class: "amount label-color" %>
-      <div class="faa-amount">
-        <%= f.text_field :amount, placeholder: '0.00', value: number_to_currency(other_income.amount), required: true %>
+      <div class="col-md-3 fa-select select-box">
+        <%#= f.select_tag :frequency, placeholder: 'HOW OFTEN *', required: true, class: "floatlabel" %>
```
Comment (scaustin34) [inline components/financial_assistance/app/views/financial_assistance/incomes/_other_income_form.html.erb:29]: if you end up back in this file, delete this

**Review by scaustin34 — CHANGES_REQUESTED** [review]
Looks good! Mostly just nitpicking the JS but i think its worth a tiny bit of refactoring

File: components/financial_assistance/app/assets/javascripts/financial_assistance/shared.js [hunk 2]
```diff
@@ -43,4 +43,34 @@ function showTemporaryMessageDiv(form_object_id, form_type) {
...
+  var warning_div = $("#date_warning_message_" + id);
+  var startDateWarning = $("#start_date_warning_" + id)
+  var endDateWarning = $("#end_date_warning_" + id)
+
+  if (warning_div.is(":visible") && !requiresStartDateWarning && !requiresEndDateWarning) {
+    warning_div.addClass('hidden');
+    return;
+  }
+
+  if (requiresStartDateWarning || requiresEndDateWarning) {
```
[the remark this answers is not in the document]
Comment (mdkaraman) [reply components/financial_assistance/app/assets/javascripts/financial_assistance/shared.js:63]: This is great!  Exactly the kind of feedback I was hoping for... thanks!

File: components/financial_assistance/app/views/financial_assistance/incomes/_other_income_form.html.erb (hunk 1 above)
Comment (mdkaraman) [reply components/financial_assistance/app/views/financial_assistance/incomes/_other_income_form.html.erb:29]: 👍

**Review by scaustin34 — DISMISSED** [review]
🤘

**Review by polographer — APPROVED** [review]
looks good!

**Pull request merged**